airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -293,6 +317,60 @@ private void reportFailure(final ConnectionUpdaterInput connectionUpdaterInput,
     }
   }
 
+  private SyncCheckConnectionFailure checkConnections(GenerateInputActivity.GeneratedJobInput jobInputs) {
+    final JobRunConfig jobRunConfig = jobInputs.getJobRunConfig();
+    final StandardSyncInput syncInput = jobInputs.getSyncInput();
+    final JsonNode sourceConfig = syncInput.getSourceConfiguration();
+    final JsonNode destinationConfig = syncInput.getDestinationConfiguration();
+    final IntegrationLauncherConfig sourceLauncherConfig = jobInputs.getSourceLauncherConfig();
+    final IntegrationLauncherConfig destinationLauncherConfig = jobInputs.getDestinationLauncherConfig();
+    SyncCheckConnectionFailure checkFailure = new SyncCheckConnectionFailure(jobRunConfig);
+
+    final int attemptCreationVersion =
+        Workflow.getVersion(CHECK_BEFORE_SYNC_TAG, Workflow.DEFAULT_VERSION, CHECK_BEFORE_SYNC_CURRENT_VERSION);
+
+    if (attemptCreationVersion < CHECK_BEFORE_SYNC_CURRENT_VERSION) {
+      // return early if this instance of the workflow was created beforehand
+      return checkFailure;
+    }
+
+    final StandardCheckConnectionInput sourceConfiguration = new StandardCheckConnectionInput().withConnectionConfiguration(sourceConfig);
+    final CheckConnectionInput checkSourceInput = new CheckConnectionInput(jobRunConfig, sourceLauncherConfig, sourceConfiguration);
+
+    if (workflowState.isResetConnection() || checkFailure.isFailed()) {
```

Comment:
I don't quite understand the ` || checkFailure.isFailed()` part here. We are creating this object several lines above here, and from what I can tell it is always constructed in a way that `isFailed()` will always be false here. Is this more just future-proofing for if we do at some point want to add anything above here that may set that to failed?

Reply:
yep - just future-proofing.